### ISSUE TITLE
[BUG] Can't use `AnimGifImage`

### ISSUE THREAD

**Issue by ImUrX** [fltk-rs/fltk-rs#1506]
## Describe the bug
I get a `Fl_Anim_GIF_Image: (null) has invalid format.` when loading a GIF image, but it works perfectly when using the same image with `GifImage`.
I get a segmentation fault when using ``AnimGifImage::load`` also, I'm using ``AnimGifImage::from_data`` to get the invalid format error

## To Reproduce
```rs
use fltk::{app, enums::FrameType, frame::Frame, image::{AnimGifImage, AnimGifImageFlags}, prelude::*, window::Window};

fn main() {
    let app = app::App::default().with_scheme(app::Scheme::Gleam);
    let mut wind = Window::new(100, 100, 400, 300, "Hello from rust");

    let mut frame = Frame::default().with_size(360, 260).center_of(&wind);
    frame.set_frame(FrameType::EngravedBox);
    let mut image = AnimGifImage::load("an-example.gif", &mut frame, ).unwrap();
    image.scale(200, 200, true, true);

    wind.make_resizable(true);
    wind.end();
    wind.show();

    app.run().unwrap();
}
```

## Expected behavior
To load the image

## Desktop info
 - OS: Arch Linux 6.6.8-arch1-1 
 - 1.4.22
 - Wayland

**Comment by MoAlyousef** [comment]
I tried your program with a local gif image on my system and it works.

Can you post the gif image as well, it could be a decoding issue upstream. If so, I'll report it.

**Comment by ImUrX** [comment]
I have used this image and it happens anyways <https://github.com/fltk/fltk/blob/master/fluid/icons/fluid.animated.gif>

Ok, so I figured stuff out, the segfault is because all this was happening in a custom widget and I wasn't storing the widgets inside the struct so their lifetime is extended.

But the ``from_data`` still gives format error, and when I use the same data bytes in `GifImage` it doesn't error.
I'm using `rust-embed` for loading the assets from the binary. Just using something like ``&Assets::get("fluid.animated.gif").unwrap().data`` should work

**Comment by MoAlyousef** [comment]
Fl_Anim_Gif_Image was relatively recently added to FLTK. It might have different lifetime requirements than other Image types which I’ll have to check out and thus modify its Drop impl or maybe report upstream if it’s an underlying bug (FLTK might be releasing it earlier than it should).

**Comment by ImUrX** [comment]
Do you have any idea why from_data fails though? I really don't understand what's happening in there 😭

**Comment by MoAlyousef** [comment]
I was able to create a repro in C++ and I've reported this here:
https://github.com/fltk/fltk/issues/897

**Comment by MoAlyousef** [comment]
I've pushed a local fix to fltk-rs. So you should be able to show an animated gif from data using master by adding the following to your Cargo.toml:

```toml
[patch.crates-io]
fltk = { git = "https://github.com/fltk-rs/fltk-rs" }
```

Once this is fixed in FLTK, I'll pull their fix.

**Comment by MoAlyousef** [comment]
This should be fixed in 1.4.23
